zilliqa/src/api/ots.rs

```diff
@@ -124,3 +126,122 @@ fn has_code(params: Params, node: &Arc<Mutex<Node>>) -> Result<bool> {
+    if reverse {
+        touched.reverse();
+    }
+
+    let mut transactions = Vec::with_capacity(page_size);
+    let mut receipts = Vec::with_capacity(page_size);
+
+    // Keep track of the current block number. Once we reach `page_size` transactions, we still need to continue adding
+    // transactions from the current block.
+    let mut current_block = u64::MAX;
+    // This will be set to false if we break out of the loop, indicating to the caller there are further pages.
+    let mut finished = true;
+
+    for hash in touched {
+        let txn = get_transaction_inner(hash, &node.lock().unwrap())
+            .unwrap()
+            .unwrap();
+
+        let txn_block_number = match txn.block_number {
+            Some(txn_block_number) => txn_block_number,
+            None => continue,
+        };
+
+        let cmp = if !reverse {
+            PartialOrd::le
+        } else {
+            PartialOrd::ge
+        };
+        if cmp(&txn_block_number, &block_number) {
+            continue;
+        }
+
+        // Don't break until we have at least `page_size` transactions AND we've added everything from the last searched block.
+        if transactions.len() >= page_size && txn_block_number != current_block {
+            finished = false;
+            break;
+        }
+
+        let timestamp = node
+            .lock()
+            .unwrap()
+            .get_block_by_hash(Hash(txn.block_hash.unwrap_or_default().0))?
+            .unwrap()
+            .timestamp();
+
+        transactions.push(txn);
+
+        let node = node.lock().unwrap();
+        let receipt = ots::TransactionReceiptWithTimestamp {
+            receipt: get_transaction_receipt_inner(hash, &node).unwrap().unwrap(),
+            timestamp: timestamp
+                .duration_since(SystemTime::UNIX_EPOCH)
+                .unwrap()
+                .as_secs(),
+        };
+        receipts.push(receipt);
+
+        current_block = txn_block_number;
+    }
+
+    // The results should always be returned in descending order (latest to earliest). If we were searching forwards
+    // in time, we should reverse the results to ensure they are in descending order.
+    if !reverse {
+        transactions.reverse();
+        receipts.reverse();
+    }
+
+    // `first_page` should be set if this was the latest page in time and `last_page` should be set if this was the
+    // earliest page in time.
+    let (first_page, last_page) = if reverse {
+        (block_number == u64::MAX, finished)
+    } else {
+        (finished, block_number == 0)
+    };
+
+    Ok(ots::Transactions {
+        transactions,
+        receipts,
+        first_page,
+        last_page,
+    })
+}
+
+fn search_transactions_after(params: Params, node: &Arc<Mutex<Node>>) -> Result<ots::Transactions> {
+    let mut params = params.sequence();
+    let address: H160 = params.next()?;
+    let block_number: u64 = params.next()?;
+    let page_size: usize = params.next()?;
+
```

Comment:
Does block_number==0 => most recent block? 
(yes, it's useless, but is it what's expected?)

Reply:
It depends on if you're searching forwards or backwards. The docs say:

> A value of 0 means you want to search from the most recent block (ots_searchTransactionsBefore) or from the genesis (ots_searchTransactionsAfter).

The actual list of transactions in the UI displays in reverse order by default, so the client makes an initial request to `ots_searchTransactionsBefore` with `block_number = 0` to get the latest transactions.

I'm not sure this is useless, so I'll leave this unresolved in case I've misunderstood you?